presto-main/src/test/java/com/facebook/presto/operator/TestHashJoinOperator.java

```diff
@@ -971,7 +1073,13 @@ private static void runDriver(ExecutorService executor, Driver driver)
     {
         executor.execute(() -> {
             if (!driver.isFinished()) {
-                driver.process();
+                try {
+                    driver.process();
+                }
+                catch (PrestoException e) {
+                    driver.getDriverContext().failed(e);
```

Comment:
Nope, it only marks that it needs distruction on the context and throws the expection up to be handled by TaskManager that is not mocked in the test.